tools/issue-labeler/backfill.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"sort"
+
+	"github.com/golang/glog"
+)
+
+type Issue struct {
+	Number int
+	Body   string
+	Labels []Label
+}
+
+type Label struct {
+	Name string
+}
+
+type IssueUpdate struct {
+	Labels []string `json:"labels"`
+}
+
+func backfill(since string, enrolledTeams map[string][]string, dryRun bool) {
+	client := &http.Client{}
+	done := false
+	page := 1
+	var issues []Issue
+	for !done {
+		url := fmt.Sprintf("https://api.github.com/repos/hashicorp/terraform-provider-google/issues?since=%s&per_page=100&page=%d", since, page)
+		req, err := http.NewRequest("GET", url, nil)
+		if err != nil {
+			glog.Exitf("Error creating request: %v", err)
+		}
+		req.Header.Add("Accept", "application/vnd.github+json")
+		req.Header.Add("Authorization", "Bearer "+os.Getenv("GITHUB_TOKEN"))
+		req.Header.Add("X-GitHub-Api-Version", "2022-11-28")
+		resp, err := client.Do(req)
+		if err != nil {
+			glog.Exitf("Error listing issues: %v", err)
+		}
+		defer resp.Body.Close()
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			glog.Exitf("Error reading response body: %v", err)
+		}
+		var newIssues []Issue
+		json.Unmarshal(body, &newIssues)
+		if len(newIssues) == 0 {
```

Comment:
I'm not sure if you've run this yet, but do we get here after attempting to request an invalid page?

Reply:
An invalid page returns an empty array.